sky/utils/controller_utils.py

```diff
@@ -767,22 +755,19 @@ def assert_no_bucket_creation(store: storage_lib.AbstractStore) -> None:
             constants.FILE_MOUNTS_WORKDIR_SUBPATH.format(run_id=run_id))
         stores = None
         if store_type is not None:
-            assert store_cls is not None
-            with sky_logging.silent():
-                stores = {
-                    store_type: store_cls(name=bucket_name,
-                                          source=workdir,
-                                          _bucket_sub_path=bucket_sub_path,
-                                          **store_kwargs)
-                }
-                assert_no_bucket_creation(stores[store_type])
+            stores = [store_type]
```

Comment:
I think there are some merge issues/conflicts in this file. E.g. this line makes no sense to me.

Reply:
I guess this is because Storage was refactored, but it makes it hard to reason about this PR

Reply:
This is intentional, as we should have the Storage object to be constructed on the remote API server side, instead on the client side. The previous way would eagerly initialize and upload local file directly on the client side, which requires cloud credentials, but now it is only initialized on the remote API server when `construct` function is called.